flake.nix

```diff
@@ -29,12 +29,26 @@
 
       lib = {
         hm = import ./modules/lib { lib = nixpkgs.lib; };
-        homeManagerConfiguration = { configuration, pkgs, lib ? pkgs.lib
-          , extraModules ? [ ], extraSpecialArgs ? { }, check ? true }@args:
+        homeManagerConfiguration = { modules ? [ ], pkgs, lib ? pkgs.lib
+          , extraSpecialArgs ? { }, check ? true
+          , configuration ? "configuration is removed, use modules instead"
```

Comment:
This seems ugly, I'd rather suggest something like `{ configuration ? null }: lib.throwIf (configuration != null) "configuration is removed, use modules instead" ...`

Reply:
Nice, i looked a little for something like `throwIf` but obviously not enough. Updated now. Btw, apologies for putting commits on your branch, you may have been planning to add something like this already. I had a bit of time to spend today and took the opportunity to try out flakes a bit.

Reply:
No worries but you did remove https://github.com/nix-community/home-manager/commit/96ebe57b47900d49e9989bc99973f991b83ca0e1. Are you planning to do that in a separate PR, or just a rebasing failure?

Reply:
It no longer seems necessary since I removed the default state version in https://github.com/nix-community/home-manager/commit/3c710201d59188754517eca0e6bd89ca34863e9c, which is already in master.